packages/thumbnail-grid-block/src/components/Thumbnail/Image.tsx

```diff
@@ -27,26 +30,52 @@ export const Image = ({
     onFilesDrop,
     id,
     onAssetChooserClick,
-}: ImageProps) => (
-    <ImageWrapper thumbnailStyles={thumbnailStyles} placeholderWrapper={!image}>
-        {image && !isLoading ? (
-            <img
-                data-test-id="thumbnail-image"
-                className="tw-object-scale-down"
-                style={thumbnailStyles.imageStyles}
-                loading="lazy"
-                src={image.genericUrl.replace('{width}', `${800 * (window?.devicePixelRatio ?? 1)}`)}
-                alt={altText || image.title || image.fileName || undefined}
-            />
-        ) : // eslint-disable-next-line unicorn/no-nested-ternary
-        isEditing ? (
-            <UploadPlaceholder
-                width={thumbnailStyles.width}
-                isLoading={isLoading}
-                openFileDialog={onOpenFileDialog}
-                onFilesDrop={(files) => onFilesDrop(files, id)}
-                openAssetChooser={onAssetChooserClick}
-            />
-        ) : null}
-    </ImageWrapper>
-);
+    appBridge,
+    isAssetViewerEnabled,
+}: ImageProps) => {
+    const { open } = useAssetViewer(appBridge);
+    const { containerWidth, setContainerRef } = useImageContainer();
+
+    const getImageComponent = () => {
+        const ImageComponent =
+            image && containerWidth ? (
+                <ResponsiveImage
+                    style={thumbnailStyles.imageStyles}
+                    testId="thumbnail-image"
+                    image={image}
+                    className="tw-object-scale-down"
+                    containerWidth={containerWidth}
+                    alt={altText || image.title || image.fileName || ''}
+                />
+            ) : null;
+
+        if (isEditing) {
+            return image && !isLoading ? (
+                ImageComponent
+            ) : (
+                <UploadPlaceholder
+                    width={thumbnailStyles.width}
+                    isLoading={isLoading}
+                    openFileDialog={onOpenFileDialog}
+                    onFilesDrop={(files) => onFilesDrop(files, id)}
+                    openAssetChooser={onAssetChooserClick}
+                />
+            );
+        } else if (image) {
+            return isAssetViewerEnabled ? (
+                <button data-test-id="thumbnail-grid-block-asset-viewer-button" onClick={() => open(image)}>
+                    {ImageComponent}
+                </button>
+            ) : (
+                ImageComponent
+            );
+        }
+        return null;
+    };
+
+    return (
+        <ImageWrapper setContainerRef={setContainerRef} thumbnailStyles={thumbnailStyles} placeholderWrapper={!image}>
+            {getImageComponent()}
+        </ImageWrapper>
+    );
```

Comment:
Implemented some parts of this solution, mainly creating more functions that render the view and edit mode parts.

Did not exactly copy this solution because of the following reasons:
1. Early return if there is no image is not expected, as in edit mode we still need to render the upload placeholder
2. There were some type errors, mainly regarding the container width.
3. In `renderEditingComponent` we need the `uploadPlaceHolder` if there is no image.
4. I try to avoid nested ternary.